packages/react-dev-utils/launchEditor.js

```diff
@@ -11,10 +11,59 @@
 const fs = require('fs');
 const path = require('path');
 const child_process = require('child_process');
+const EventEmitter = require('events').EventEmitter;
 const os = require('os');
 const chalk = require('chalk');
 const shellQuote = require('shell-quote');
 
+// Inspired by https://github.com/rannn505/node-powershell
+const EOI = 'EOI';
+class PowerShell extends EventEmitter {
+  constructor() {
+    super();
+
+    this._proc = child_process.spawn(
+      'powershell.exe',
+      ['-NoLogo', '-NoProfile', '-NoExit', '-Command', '-'],
+      {
+        stdio: 'pipe',
+      }
+    );
+
+    this._proc.on('error', () => {
+      // Needs to be registered... already catched by if-statement below
+    });
+
+    if (!this._proc.pid) {
+      throw new Error('Failed to start PowerShell');
+    }
+
+    let output = [];
+    this._proc.stdout.on('data', data => {
+      if (data.indexOf(EOI) !== -1) {
+        this.emit('resolve', output.join(''));
+        output = [];
+      } else {
+        output.push(data);
+      }
+    });
+  }
+
+  invoke(cmd) {
+    return new Promise(resolve => {
+      this.on('resolve', data => {
+        resolve(data);
+        this.removeAllListeners('resolve');
```

Comment:
Are there any possible race conditions here? What if we call `invoke` twice before it resolves? Will both be resolved correctly?

Reply:
Uhm. I will test that next weekend 🙂

Reply:
Fixed 🙂